pkg/arm/data/master-startup.sh

```diff
@@ -255,7 +255,7 @@ oauthConfig:
       kind: HTPasswdPasswordIdentityProvider
   masterCA: ca.crt
   masterPublicURL: {{ print "https://" .ContainerService.Properties.OrchestratorProfile.OpenShiftConfig.PublicHostname | quote }}
-  masterURL: {{ print "https://" .ContainerService.Properties.OrchestratorProfile.OpenShiftConfig.PublicHostname | quote }}
+  masterURL: {{ print "https://" .ContainerService.Properties.MasterProfile.FQDN | quote }}
```

Comment:
I think we should have the RP set PublicHostname to FQDN if it wasn't set, and here we should continue using PublicHostname.
But in the master cert generation (for example), there should be SANs for both PublicHostname and FQDN, if they differ.

Reply:
PublicHostname, if set is in user control. Lets say FQDN is generated as `mjudeikis-9a4dc255.osa.westus.azmk8s.io` and user want to have `console.openshift.example.com` as the main URL. The user can set this on creation in PublicHostname and we will return FQDN and say "create a cname now please". In addition, if the user will want to have proper valid certificates at any point in future (question if this is even for GA, those values cant be set to the same value).

```
masterURL and masterPublicURL cannot be set to the same value when using custom named certificates. Both values need to be resolvable to the load balancer, but if masterURL is the same as masterPublicURL then hosts within the cluster will try to verify against the custom named certificates.

The masterURL needs to be resolvable by the nodes internal to the cluster; thus it does not need to be resolvable outside the cluster, whereas masterPublicURL must be externally resolvable.
```

Reply:
TODO: check customer_cert stuff https://docs.openshift.com/container-platform/3.5/install_config/certificate_customization.html

Reply:
TODO: Test if we can change    PublicHostname post fact